ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -186,15 +186,22 @@ private IndexValue(long size, Offset offset, byte flags, long expiresAtMs, long
     this.size = size;
     this.offset = offset;
     this.flags = flags;
-    int expiryInSecs = (int) (expiresAtMs / Time.MsPerSec);
-    this.expiresAtMs = expiresAtMs != Utils.Infinite_Time ? expiryInSecs * Time.MsPerSec : Utils.Infinite_Time;
+    this.expiresAtMs = getTimeInMsToTheNearestSec(expiresAtMs);
     this.originalMessageOffset = originalMessageOffset;
-    int operationTimeInSecs = (int) (operationTimeInMs / Time.MsPerSec);
-    this.operationTimeInMs =
-        operationTimeInMs != Utils.Infinite_Time ? operationTimeInSecs * Time.MsPerSec : Utils.Infinite_Time;
+    this.operationTimeInMs = getTimeInMsToTheNearestSec(operationTimeInMs);
     this.serviceId = serviceId;
     this.containerId = containerId;
-    version = PersistentIndex.VERSION_1;
+    version = PersistentIndex.CURRENT_VERSION;
```

Comment:
Nope. The idea here is that, any new IndexSegment and IndexValue should take it the recent version of PersistentIndex and hence this way.